Revert abort controller and more sophisticated nodejs env detection (#401)
fauna/faunadb-js: README.md

})
```

> ## Knowing issues

Reviewer: ## Known issues

### Using with Cloudflare Workers

Cloudflare Workers has neither XMLHttpRequest nor fetch in global scope.
Therefore, cross-fetch package is unable to inject fetch, and throws an error.
fetch function is injected via closure, so the workaround would be to pass
the fetch objects when initiating the FaunaDB client config. Cloudflare also 
doesn't support AbortController, which terminates requests as well as streams. 
Here is a workaround:

Reviewer: just some minor grammatical edits

Author: done

})
```

## Known issues

### Using with Cloudflare Workers

> Cloudflare Workers has neither XMLHttpRequest nor fetch in global scope.

Reviewer 2: ```suggestion
Cloudflare Workers have neither XMLHttpRequest nor fetch in the global scope.
```

> Therefore, cross-fetch package is unable to inject fetch, and throws an error.

Reviewer 2: ```suggestion
Therefore, the `cross-fetch` package is unable to inject its own `fetch()` function, and throws an error.
```

> fetch function is injected via closure, so the workaround would be to pass

Reviewer 2: ```suggestion
The `fetch()` function is injected via a closure, so the workaround would be to pass
```

> the fetch objects when initiating the FaunaDB client config. Cloudflare also

Reviewer 2: Objects? Do you mean "function"? If not, it would be good to provide a bit more context here.

```suggestion
the fetch objects when initiating the FaunaDB client config. Cloudflare Workers also
```

Author: yes, it's about function

Reviewer 2: That means that `objects` needs to be replaced with `function`.

> doesn't support AbortController, which terminates requests as well as streams.

Reviewer 2: ```suggestion
doesn't support the use of an AbortController, which terminates requests as well as streams.
```
When you say "which terminates", the subject is ambiguous. Does an AbortController terminate requests/streams, or does the lack of an AbortController cause that?

Author: AbortController terminates requests. Cloudflare doesn't have AbortController. so even if a user decides to close a stream, it would be still alive

Reviewer 2: Thanks for the information. I'm still not completely certain that I know how to write the last sentence to resolve the ambiguity; let me know if this sounds correct:

`... the use of an AbortController, which can cause request and stream termination without sufficient error handling.`

Here is a workaround:

```javascript
const c = new faunadb.Client({
  secret: 'your secret',
  fetch: (url, params) => {
    const signal = params.signal

>     delete params.signal

Reviewer 3: why `delete` is necessary here?

Author: Cloudflare doesn't implement signal into their fetch, if the signal property is not removed, an error would be thrown

    const abortPromise = new Promise(resolve => {
      if (signal) {
        signal.onabort = resolve
      }
    })
    return Promise.race([abortPromise, fetch(url, params)])
  },

> })

Reviewer 2: Nice! I haven't seen that done before.

>     return Promise.race([abortPromise, fetch(url, params)])

Reviewer 3: Where this `fetch` come from? Shouldn't we include the require statements in the snippet?

Author: no, Cloudflare inject it into a function handler
